core-web/libs/sdk/client/src/lib/client/content/shared/utils.ts

```diff
@@ -26,3 +26,62 @@ export function sanitizeQueryForContentType(query: string, contentType: string):
             : original; // Return the field if it is a content type field
     });
 }
+
+/**
+ * @description
+ * Determines whether a site ID constraint should be added to a query based on existing constraints.
+ *
+ * The site ID constraint is added only when:
+ * - Query doesn't already contain a positive site constraint (+conhost)
+ * - Query doesn't explicitly exclude the specified site ID (-conhost:siteId)
+ * - Site ID is provided and configured
+ *
+ * @example
+ * ```ts
+ * const query = '+contentType:Blog +languageId:1';
+ * const siteId = '123';
+ * const shouldAdd = shouldAddSiteIdConstraint(query, siteId); // true
+ * ```
+ *
+ * @example
+ * ```ts
+ * const query = '+contentType:Blog -conhost:123';
+ * const siteId = '123';
+ * const shouldAdd = shouldAddSiteIdConstraint(query, siteId); // false (explicitly excluded)
+ * ```
+ *
+ * @example
+ * ```ts
+ * const query = '+contentType:Blog +conhost:456';
+ * const siteId = '123';
+ * const shouldAdd = shouldAddSiteIdConstraint(query, siteId); // false (already has constraint)
+ * ```
+ *
+ * @export
+ * @param {string} query - The Lucene query string to analyze
+ * @param {string | number | null | undefined} siteId - The site ID to check for
+ * @returns {boolean} True if site ID constraint should be added, false otherwise
+ */
+export function shouldAddSiteIdConstraint(
+    query: string,
+    siteId: string | number | null | undefined
+): boolean {
+    // No site ID configured
+    if (!siteId) {
+        return false;
+    }
+
+    // Query already contains a positive site constraint
+    const hasExistingSiteConstraint = query.includes('+conhost');
```

Comment:
It's safer to use a RegExp `/\+conhost/gi.exec(query)`

Reply:
the `i` flag will cover any case sensitive case, i think on lucene query it also works with `conHost`